### PR TITLE
Adding 

### PR DESCRIPTION
**Pull request by Alona-Kirichenko**
-  adding Yup and useForm libraries to form and validate input fields.
- making input field compost adapted to Yup and useForm
- making some styles as separate file 
-  I was starting with refactoring to TypesScript but understood that it would take a huge amount of time. I took off all changes except of file "types", there I wrote object types the way we use it in front end, so we can use it in future

### CODE REVIEW

File: src/android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
```
Comment (covetous) [inline src/android/gradle/wrapper/gradle-wrapper.properties:3]: Nedgradera till äldre version?

File: src/ios/DoGoodGetGood.xcworkspace/contents.xcworkspacedata [hunk 2]
```diff
@@ -7,4 +7,7 @@
    <FileRef
       location = "group:Pods/Pods.xcodeproj">
    </FileRef>
+   <FileRef
+      location = "group:../../../../DO_GOOD_GET_GOOD/do-good-get-good/src/ios/DoGoodGetGood.xcodeproj">
+   </FileRef>
```
Comment (covetous) [inline src/ios/DoGoodGetGood.xcworkspace/contents.xcworkspacedata:12]: Det här är en dublett i filen och pathen är bara mög. Så återställ den här filen.

File: src/ios/Podfile [hunk 3]
```diff
@@ -7,6 +7,7 @@ require Pod::Executable.execute_command('node', ['-p',
 
 platform :ios, min_ios_version_supported
 prepare_react_native_project!
+project '/Users/alonakirichenko/DO_GOOD_GET_GOOD/do-good-get-good/src/ios/DoGoodGetGood.xcodeproj'
```
Comment (covetous) [inline src/ios/Podfile:10]: NO!

File: src/navigate.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { CreateActivity } from "./screens/CreateActivity";
 import { ActivityCard } from "./screens/ActivityCard";
 import { ImagesGallery } from "./screens/ImagesGallery";
 import CreateUser from "./screens/CreateUser";
-import ChangeUser from "./screens/ChangeUser";
+import ChangeUser from "./screens/ChangeUser/ChangeUser";
```
Comment (covetous) [inline src/navigate.js:11]: Det här ser också konstigt ut?

File: src/screens/ChangeUser/ChangeUser.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<UserName>({
+    defaultValues: { name: userName, surname: userSurname },
+    resolver: yupResolver(schema),
+  });
+  const [changedStatus, setcChangedStatus] = useState(statusActive);
```
Comment (Mattan1999) [inline src/screens/ChangeUser/ChangeUser.tsx:56]: Fix spelling to `setChangedStatus`

File: src/screens/ChangeUser/ChangeUser.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          error={errors.surname}
+          name={"surname"}
+        />
+
+        <TouchableOpacity
+          style={{
+            marginTop: 10,
+            alignSelf: "flex-start",
+          }}
+          onPress={() => setcChangedStatus(!changedStatus)}
```
Comment (Mattan1999) [inline src/screens/ChangeUser/ChangeUser.tsx:95]: Same here, fix spelling to `setChangedStatus`

File: src/styles/borderStyle.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { StyleProp } from "react-native/Libraries/StyleSheet/StyleSheet";
+import colors from "../assets/theme/colors";
+import {
+  FlexStyle,
+  TextStyle,
+} from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+export const borderStyle = (
+  isEror: boolean = false,
```
Comment (Mattan1999) [inline src/styles/borderStyle.ts:8]: Fix spelling `isError`

File: src/styles/borderStyle.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+): StyleProp<FlexStyle | TextStyle> => ({
+  borderRadius: 5,
+  borderWidth: 1,
+  borderColor: isEror ? colors.error : colors.background,
```
Comment (Mattan1999) [inline src/styles/borderStyle.ts:12]: Same here, fix spelling `isError`

File: src/utilily/types.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+export type Concept = {
+  id?: string;
+  body: string;
+  heading: string;
+  orderID: number;
+};
+
+export type TimeEntrie = {
```
Comment (Mattan1999) [inline src/utilily/types.ts:21]: Fix spelling to `TimeEntry`

File: src/utilily/types.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  userID?: User["id"];
+  title: string;
+  photo?: string;
+  city?: string;
+  date: Date;
+  statusConfirmed?: boolean;
+  time: number;
+};
+
+export type AtivitiesAndAccumulatedTime = {
```
Comment (Mattan1999) [inline src/utilily/types.ts:34]: Fix spelling to `ActivitiesAndAccumulatedTime`

File: src/utilily/types.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  accumulatedTime: number;
+  activityID: Activity["id"];
+};
+
+export type User = {
+  id: string;
+  activitiesAndAccumulatedTime?: Array<AtivitiesAndAccumulatedTime>;
```
Comment (Mattan1999) [inline src/utilily/types.ts:41]: Fix spelling to `ActivitiesAndAccumulatedTime`

File: src/utilily/types.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  adminID?: User["id"];
+  connectedActivities?: Array<Activity["id"]>;
+  firstName: string;
+  lastName: string;
+  email?: string;
+  role?: string;
+  timeEntries?: TimeEntrie | Array<TimeEntrie>;
```
Comment (Mattan1999) [inline src/utilily/types.ts:48]: Fix spelling to TimeEntry

File: src/utilily/utils.ts
Comment (Mattan1999) [inline src/utilily/utils.ts:1]: Delete if the file is empty.

File: src/Gemfile [hunk 13]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 # You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
-ruby ">= 2.6.10"
```
Comment (Mattan1999) [inline src/Gemfile:4]: Change this back since the latest react native version is still using `>= 2.6.10`.

File: src/components/InputField.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+              onBlur={onBlur}
+              onChangeText={(value) => onChange(value)}
+              value={value}
+              placeholder={placeholderText}
+              placeholderTextColor={colors.dark}
+              returnKeyType={returnKeyType}
+              keyboardType={keyboardType}
+              textContentType={textContentType}
+              secureTextEntry={secureTextEntry}
+            />
```
Comment (Mattan1999) [inline src/components/InputField.tsx:60]: Please change so that there is no auto capitalisation for e-mail inputs.

File: src/Gemfile (hunk 13 above)
Comment (Alona-Kirichenko) [reply src/Gemfile:4]: I got an error that this is too old and I had to update it

**Review by Mattan1999 — APPROVED** [review]

**Pull request merged**